### PR TITLE
chore/kotlin_upgrade_roundup

### PR DESCRIPTION
**Pull request by rodvar**
- contribs to #1163 
 - replace usages of kotlinx.datetime.Instant with kotlin.time.Instant 
 - other minor warnings fixes related to recent Kotlin upgrade

## Summary by CodeRabbit

* **Refactor**
  * Migrated internal date-time type usages across the app.
  * Updated dropdown anchor implementation used by the UI.
  * Improved iOS platform behavior: device language detection no longer falls back to "en", and image decoding no longer throws on decode failure.

* **Bug Fixes**
  * Minor formatting/placeholder handling for message formatting and BTC sats display to avoid null/coercion edge cases.

* **Tests**
  * Aligned unit tests with the updated types and utilities.

### CODE REVIEW

File: shared/domain/src/iosMain/kotlin/network/bisq/mobile/data/utils/PlatformDomainAbstractions.ios.kt [hunk 1]
```diff
@@ -116,7 +116,7 @@ actual fun encodeURIParam(param: String): String =
         .stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLPathAllowedCharacterSet)
         ?: param
 
-actual fun getDeviceLanguageCode(): String = NSLocale.currentLocale.languageCode ?: "en"
+actual fun getDeviceLanguageCode(): String = NSLocale.currentLocale.languageCode
```
[the remark this answers is not in the document]
Comment (rodvar) [reply shared/domain/src/iosMain/kotlin/network/bisq/mobile/data/utils/PlatformDomainAbstractions.ios.kt]: The Kotlin compiler itself told us languageCode is non-nullable String (that's why we got the "elvis always returns left operand" warning). The expect contract returns String, and the K/N mapping of NSLocale.languageCode is non-null.
